azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -606,6 +912,112 @@ func flattenEventGridEventSubscriptionSubjectFilter(filter *eventgrid.EventSubsc
+	stringContains := make([]interface{}, 0)
+	stringIn := make([]interface{}, 0)
+	stringNotIn := make([]interface{}, 0)
+
+	for _, item := range *input {
+		switch f := item.(type) {
+		case eventgrid.BoolEqualsAdvancedFilter:
+			boolEquals = append(boolEquals, map[string]interface{}{
+				"key":   *f.Key,
+				"value": *f.Value,
+			})
+		case eventgrid.NumberGreaterThanAdvancedFilter:
+			numberGreaterThan = append(numberGreaterThan, map[string]interface{}{
+				"key":   *f.Key,
+				"value": *f.Value,
+			})
+		case eventgrid.NumberGreaterThanOrEqualsAdvancedFilter:
+			numberGreaterThanOrEquals = append(numberGreaterThanOrEquals, map[string]interface{}{
+				"key":   *f.Key,
+				"value": *f.Value,
+			})
+		case eventgrid.NumberLessThanAdvancedFilter:
+			numberLessThan = append(numberLessThan, map[string]interface{}{
+				"key":   *f.Key,
+				"value": *f.Value,
+			})
+		case eventgrid.NumberLessThanOrEqualsAdvancedFilter:
+			numberLessThanOrEquals = append(numberLessThanOrEquals, map[string]interface{}{
+				"key":   *f.Key,
+				"value": *f.Value,
+			})
+		case eventgrid.NumberInAdvancedFilter:
+			numberIn = append(numberIn, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		case eventgrid.NumberNotInAdvancedFilter:
+			numberNotIn = append(numberNotIn, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		case eventgrid.StringBeginsWithAdvancedFilter:
+
+			stringBeginsWith = append(stringBeginsWith, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		case eventgrid.StringEndsWithAdvancedFilter:
+
+			stringEndsWith = append(stringEndsWith, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		case eventgrid.StringContainsAdvancedFilter:
+			stringContains = append(stringContains, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		case eventgrid.StringInAdvancedFilter:
+			stringIn = append(stringIn, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		case eventgrid.StringNotInAdvancedFilter:
+			stringNotIn = append(stringNotIn, map[string]interface{}{
+				"key":    *f.Key,
+				"values": *f.Values,
+			})
+		}
+	}
+
+	filters := map[string][]interface{}{
+		"bool_equals":                   boolEquals,
+		"number_greater_than":           numberGreaterThan,
+		"number_greater_than_or_equals": numberGreaterThanOrEquals,
+		"number_less_than":              numberLessThan,
+		"number_less_than_or_equals":    numberLessThanOrEquals,
+		"number_in":                     numberIn,
+		"number_not_in":                 numberNotIn,
+		"string_begins_with":            stringBeginsWith,
+		"string_ends_with":              stringEndsWith,
+		"string_contains":               stringContains,
+		"string_in":                     stringIn,
+		"string_not_in":                 stringNotIn,
+	}
+
+	results = append(results, filters)
+
+	return results
```

Comment:
(not an issue, just for info) since we're hard-coding a single element here, we can reduce this to:

```suggestion
	return []interface{}{
		map[string][]interface{}{
 			"bool_equals":                   boolEquals,
 			"number_greater_than":           numberGreaterThan,
 			"number_greater_than_or_equals": numberGreaterThanOrEquals,
 			"number_less_than":              numberLessThan,
 			"number_less_than_or_equals":    numberLessThanOrEquals,
 			"number_in":                     numberIn,
 			"number_not_in":                 numberNotIn,
 			"string_begins_with":            stringBeginsWith,
 			"string_ends_with":              stringEndsWith,
 			"string_contains":               stringContains,
 			"string_in":                     stringIn,
 			"string_not_in":                 stringNotIn,
 		}
	}
```